### ISSUE TITLE
[enhancement]: Metadata node rework - Streamline and avoids iterate andcollect node issues

### ISSUE THREAD

**Issue by skunkworxdark** [invoke-ai/InvokeAI#5107]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Contact Details

skunkworxdark

### What should this feature add?

As already discussed with  @psychedelicious on discord.  Update the Metadata collection nodes to make it more streamlined and avoid use of the `collect` node, which currently causes issues if it while also using an `iterate` node as it collects everything over all iterations not just the current one.

Alter the `Metadata Item` by adding an optional MetadataDict as an input. It would add/update the MetadataDict with the passed in label & value, If no MetadataDict is passed in it just creates a new one and passes it out. The `Metadata Item` nodes can be chained together one `Metadata Item` to another inside an iterate node.

Additionally a `Metadata Join` node would replace the current `Merge Metadata` node. This takes two MetadataDicts and combines them with the second one replacing any duplicates in the first. This would avoid using the `collect` node again that the existing `Merge Metadata` needs.

This approach also removes the need for the final Metadata Invocation as the data is always in a MetadataDict.   At each stage you would need to check if the label already exists and update it if it did but that should be minor and the same in the Join node with a simple rule that the second Dict would override the first if you have duplicate labels.

**Comment by skunkworxdark** [comment]
Could we also have a node to pull metadata, both freeform and core metadata, out of an image?

**Comment by psychedelicious** [comment]
> Could we also have a node to pull metadata, both freeform and core metadata, out of an image?

Does the service method to get metadata not suffice? Eg `context.services.images.get_metadata(image_name)`

**Comment by skunkworxdark** [comment]
Yes it probably does I've not tried it yet. However it needs to be in a node to use it in a workflow.  As this is a enhancement request to improve the metadata nodes I thought adding some nodes to pull out metadata would fit in with the original request.  It feels to me like this kind of functionality should be available on core invoke rather than in a community node.

One use case for this kind of metadata retreval node is that you generate some images and then at a later time you want to reprocess  them. This would allow you to automatically pull out settings, prompts etc for the image to use in the reprocess workflow.
